src/main/java/bio/terra/janitor/db/JanitorDao.java

```diff
@@ -139,31 +169,37 @@ public void createResource(TrackedResource resource, Map<String, String> labels)
   }
 
   /** Retrieve the {@link CleanupFlight}s associated with the tracked resource. */
-  public List<CleanupFlight> getFlights(TrackedResourceId trackedResourceId) {
+  @Transactional(propagation = Propagation.SUPPORTS)
+  public List<CleanupFlight> retrieveFlights(TrackedResourceId trackedResourceId) {
     String sql =
         "SELECT flight_id, flight_state from cleanup_flight WHERE tracked_resource_id = :tracked_resource_id;";
     MapSqlParameterSource params =
         new MapSqlParameterSource().addValue("tracked_resource_id", trackedResourceId.uuid());
     return jdbcTemplate.query(sql, params, CLEANUP_FLIGHT_ROW_MAPPER);
   }
 
-  public Optional<CleanupFlightState> getFlightState(String flightId) {
+  @Transactional(propagation = Propagation.SUPPORTS)
+  public Optional<CleanupFlightState> retrieveFlightState(String flightId) {
     String sql = "SELECT flight_state from cleanup_flight WHERE flight_id = :flight_id;";
     MapSqlParameterSource params = new MapSqlParameterSource().addValue("flight_id", flightId);
     String rawState =
         DataAccessUtils.singleResult(jdbcTemplate.queryForList(sql, params, String.class));
     return Optional.ofNullable(rawState).map(CleanupFlightState::valueOf);
   }
 
-  /** Modifies the {@link CleanupFlightState} of a single flight. */
-  public void setFlightState(String flightId, CleanupFlightState flightState) {
+  /**
+   * Modifies the {@link CleanupFlightState} of a single flight. Returns whether there was a flight
+   * with the flightId.
+   */
+  @Transactional(propagation = Propagation.SUPPORTS)
+  public boolean updateFlightState(String flightId, CleanupFlightState flightState) {
```

Comment:
the updateTrackedResource method you returned Optional<TrackedResource>, but here, you only returns boolean. Do we want to have some consistent guideline about the return for an update operation?

Reply:
There's `retrieveTrackedResource` that doesn't modify and returns an Optional.
There's `updateFlightState`that does modify and returns a boolean.
There's `updateResourceForCleaning` that does modify and returns an Optional. I'm going to refactor it to not have the logic and not do an update that returns the optional, so yes.
Does that sg?